### PR TITLE
[metrics][user] Add custom metrics to the user service

### PR DESCRIPTION
**Pull request by jaylees14**
- Add some custom metrics to the user service
- Show success and failure rate charts, averaged over the last 5 mins

The way in which you add a failure metric is kinda annoying, there's lots of repeated code - I'm gonna pull this out into a utility function before adding any more.

Also apologies for the JSON.. looks like I've just found what I'm generating for the next few days ❤️ 

Currently looks like:

Screenshot 2020-03-18 at 16 20 06

### CODE REVIEW

File: user/metrics.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package main
```
Comment (jaylees14) [inline user/metrics.go]: Not sure if this should go in a separate package @smithwjv?

Comment (willjvsmith) [reply user/metrics.go]: Yeah stick it in a new one I reckon

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀 

Surprised no other logs were recommended. Perhaps database accesses? Or when u get round to Match, interservice-comms would be a good one

**Comment by jaylees14** [comment]
> LGTM 🚀
> 
> Surprised no other logs were recommended. Perhaps database accesses? Or when u get round to Match, interservice-comms would be a good one

Oh this isn't by all means all of them - just a couple of easy ones to get us started 👍

**Review by c-harding — APPROVED** [review]

**Pull request merged**